src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/Pipeline.cs

```diff
@@ -0,0 +1,498 @@
+
+            if (data == null)
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "TensorsData is invalid");
+
+            NNStreamerError ret = NNStreamerError.None;
+
+            /* Get src node handle */
+            NodeHandleInfo hInfo;
+            if (_nodeHandleList.ContainsKey(srcNodeName))
+            {
+                hInfo = _nodeHandleList[srcNodeName];
+            }
+            else
+            {
+                IntPtr srcHandle = IntPtr.Zero;
+                ret = Interop.Pipeline.GetSrcHandle(_handle, srcNodeName, out srcHandle);
+                NNStreamer.CheckException(ret, "Failed to get Source Node handle");
+
+                hInfo = new NodeHandleInfo(srcHandle, HandleType.Source);
+                _nodeHandleList.Add(srcNodeName, hInfo);
+            }
+
+            /* Prepare TensorsData */
+            data.PrepareInvoke();
+
+            /* Input data */
+            ret = Interop.Pipeline.InputSrcData(hInfo.Handle, data.GetHandle(), PipelineBufferPolicy.NotFreed);
+            NNStreamer.CheckException(ret, "Failed to input tensors data to source node: " + srcNodeName);
+        }
+
+        /// <summary>
+        /// Controls the valve. Set the flag true to open, false to close.
+        /// </summary>
+        /// <param name="valveNodeName">The name of valve node</param>
+        /// <param name="open">The flag to control the flow (True for open)</param>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void ControlValve(string valveNodeName, bool open)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            /* Check the parameter */
+            if (string.IsNullOrEmpty(valveNodeName))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "Node Name is invalid");
+
+            /* Get valve node handle */
+            NNStreamerError ret = NNStreamerError.None;
+            NodeHandleInfo hInfo;
+            if (_nodeHandleList.ContainsKey(valveNodeName))
+            {
+                hInfo = _nodeHandleList[valveNodeName];
+            }
+            else
+            {
+                IntPtr valveHandle = IntPtr.Zero;
+                ret = Interop.Pipeline.GetValveHandle(_handle, valveNodeName, out valveHandle);
+                NNStreamer.CheckException(ret, "Failed to get Valve Node handle: " + valveNodeName);
+
+                hInfo = new NodeHandleInfo(valveHandle, HandleType.Valve);
+                _nodeHandleList.Add(valveNodeName, hInfo);
+            }
+
+            /* Set valvue status */
+            ret = Interop.Pipeline.OpenValve(hInfo.Handle, open);
```

Comment:
I think, NodeHandleInfo class should has `OpenValve` method
``` c#
hInfo.OpenValve(open);
```

Reply:
added new class for each node (switch/valve/source)